src/comment.rs

```diff
@@ -81,11 +81,11 @@ pub fn rewrite_comment(orig: &str,
             let rewrite = try_opt!(rewrite_string(line, &fmt));
             result.push_str(&rewrite);
         } else {
-            if line.len() == 0 {
-                result.pop(); // Remove space if this is an empty comment.
-            } else {
-                result.push_str(line);
+            if line.len() == 0 || line.starts_with('!') {
```

Comment:
We're already ignoring the `//` here, right? So why special case doc comments? In the test it looks like we still format `//! foo`, so I'm not clear what this is doing. Should we also consider `///` comments here?